langkit/templates/pkg_init_body_ada.mako

```diff
@@ -19,6 +20,14 @@ package body ${ada_lib_name}.Init is
       if not GNATCOLL.Iconv.Has_Iconv then
          raise Program_Error with "Libiconv is not available";
       end if;
+
+      --  We'll also initialize GNATCOLL.Traces here. It seems preferable to do
+      --  this here rather than in elaboration: Ada API users won't be forced
+      --  to use the config file, if they're GNATCOLL.Traces users with a
+      --  different workflow. However, everybody else will (Python, C, OCaml
+      --  API users), which seems like a good compromise.
+
+      GNATCOLL.Traces.Parse_Config_File;
```

Comment:
Actually, the fact that our current Ada programs don’t call this procedure is an oversight. I have a commit locally to fix that, making the call to `Initialize` on elaboration. The documentation of this public API already says that it must be called before anything else in LAL/Langkit_Support, so preventing multiple calls to Initialize looks like a breaking API change. What about creating a new API entry point instead?

Reply:
Another reason we would want to create a dedicated entry point in bindings for this is that this will play badly with Ada programs embedding a Python interpreter that loads LAL (for instance GS): the call to Parse_Config_File will conflict with GS’.

Reply:
> Actually, the fact that our current Ada programs don’t call this procedure is an oversight. I have a commit locally to fix that, making the call to Initialize on elaboration. 

I don't understand what you mean. It's definitely not an oversight that it is not called during the elaboration of Libadalang, I considered it and I don't like this solution. `GNATCOLL.Traces` might be used in client applications that don't use config files, and that might create confusing situations.

> The documentation of this public API already says that it must be called before anything else in LAL/Langkit_Support, so preventing multiple calls to Initialize looks like a breaking API change. What about creating a new API entry point instead?

I don't understand this comment either. The call to `Parse_Config_File` can be done multiple times without any problems, what would make you think otherwise?

Reply:
> Another reason we would want to create a dedicated entry point in bindings for this is that this will play badly with Ada programs embedding a Python interpreter that loads LAL (for instance GS): the call to Parse_Config_File will conflict with GS’.

Hmm, indeed, it might be called with different parameters in GNATStudio. However, adding a new entry point in bindings is way more work than I'm ready to put into this given that it's only of theoretical importance at the moment